R/module_nested_tabs.R

```diff
@@ -243,7 +243,7 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
     }
 
     if (is_arg_used(modules$server, "data")) {
-      data <- .datasets_to_data(modules, datasets, trigger_data)
+      data <- reactive(.datasets_to_data(modules, datasets))
```

Comment:
`trigger_data()` is one of the most important lines in teal. It triggers data only when module is visible. Please read comments (219-222) and see the code (223-233).
 
```suggestion
      data <- eventReactive(trigger_data(), .datasets_to_data(modules, datasets))
```